src/core/net/nat64_translator.cpp

```diff
@@ -445,7 +447,8 @@ Error Translator::TranslateIcmp6(Message &aMessage)
 
 Error Translator::SetIp4Cidr(const Ip4::Cidr &aCidr)
 {
-    Error err = kErrorNone;
+    Error err          = kErrorNone;
+    State initialState = GetState();
```

Comment:
How about caching the state when it's changed, instead of calculating the state every time?